src/Mvc/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets

```diff
@@ -16,15 +16,24 @@
   </ItemGroup>
 
   <Target Name="OpenApiGetDocuments" Returns="@(_OpenApiProjectDocuments)">
+    <Error Text="OpenAPI document generation is disabled. Add '<OpenApiGenerateDocuments>true</OpenApiGenerateDocuments>' to the project."
+        Condition=" '$(OpenApiGenerateDocuments)' != 'true' " />
+
     <ReadLinesFromFile File="$(_OpenApiDocumentsCache)">
       <Output TaskParameter="Lines" ItemName="_OpenApiProjectDocuments" />
     </ReadLinesFromFile>
+
+    <Warning Text="Application does not have any registered documents. Update its 'Startup' class to register a document."
+        Condition=" '@(_OpenApiProjectDocuments)' == '' " />
   </Target>
 
   <Target Name="GenerateOpenApiDocuments"
       Condition=" '$(OpenApiGenerateDocuments)' == 'true' "
       Inputs="$(TargetPath)"
       Outputs="$(_OpenApiDocumentsCache)">
+    <Error Text="OpenAPI document generation is not supported when targeting netcoreapp2.0 or earlier. Disable the feature or move to a later target framework."
+        Condition=" '$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1' " />
```

Comment:
Where is `_TargetFrameworkVersionWithoutV` defined? Also, do we support this in all desktop projects?

Reply:
In Microsoft.NET.Build.Extensions.targets of the .NET SDK. This also lands in MSBuild\Current\Microsoft.Common.targets\ImportAfter\Microsoft.NET.Build.Extensions.targets within your VS installation (for use with full `msbuild`).

Reply:
Since the property is private to the SDK, we shouldn't rely on it.